add `json()` impl to `Request`

minauteur opened the pull request:
In draft until feedback on approach is given in issue #199 thread

hamza1311 commented:
Sorry, it took me a little bit to get to this. It looks good. You can go ahead and finish it up so it can be merged

minauteur commented:
No worries!
I'll get this cleaned up and have tests in order so this can be ready for review sometime this evening or tomorrow morning.
Thanks!

minauteur commented:
I've cleaned up the formatting, and--for now--I've updated the test `post_json()` in `crates/net/tests/http.rs` to utilize the convenience method added in the PR--passes locally with `HTTPBIN_URL` and `ECHO_SERVER_URL` environment variables set accordingly.
Happy to flesh out tests in a later PR if this is sufficient, otherwise, I may need a some guidance on where (an) added test(s) should live for covering this method specifically.

minauteur commented:
OK--I've added a separate test: `json_body()` to assert that we get a `Result::Ok` back from `Request::json()` when we pass it a serializable type.
I also updated `post_json()` to use the added method for creating its requests.
An additional field: `num: i16` has also been added to the structs used for both tests.
@hamza1311 let me know if you'd like me to add or amend anything--happy to help!

Thanks again for the feedback!f

hamza1311 commented on crates/net/src/http/mod.rs:
```diff
@@ -124,7 +125,11 @@ impl Request {
         self
     }
-
+    /// convenience method for json request bodies
```
```suggestion
    /// A convenience method to set JSON as request body
    ///
    /// # Note
    /// 
    /// This method also sets the `Content-Type` header to `application/json`
```

hamza1311 reviewed (changes requested):
Looks good. Thanks for the PR!

Just need to update the documentation and this is good to go

minauteur commented:
✔️ Thanks so much!

hamza1311 commented:
Thanks for the PR. This is good to go

minauteur commented:
Awesome--feel free to merge whenever!
Not sure if the comment needs that line deleted from the linting check--happy to do that if so

hamza1311 commented:
Please ensure that the CI is green

minauteur commented:
> Please ensure that the CI is green

Thanks.
Looks like you (or another maintainer) will need to approve running the workflow, but it should pass with the latest commit!
